app/scripts/modules/core/src/artifact/expectedArtifactSelector.component.ts

```diff
@@ -0,0 +1,49 @@
+import { copy, IComponentOptions, IController, module } from 'angular';
+
+import { IArtifact, IExpectedArtifact } from 'core/domain';
+
+class ExpectedArtifactSelectorCtrl implements IController {
+  public command: any;
+  public idField: string;
+  public expectedArtifacts: IExpectedArtifact[];
+
+  public summarizeExpectedArtifact(expected: IExpectedArtifact): string {
```

Comment:
This logic could be implemented in a custom Angular filter (if this component doesn't fit perfectly in Jake's todo).

Reply:
Jake can drop this in right here, I don't have a env to test it comfortably & we don't have integration tests for this yet so I'm nervous to make the change myself.